File: hileusesimple/src/main/java/com/xiongtao/hileusesimple/simple1/ui/MainActivity.java
package com.xiongtao.hileusesimple.simple1.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.xiongtao.hileusesimple.R;
import com.xiongtao.hileusesimple.simple1.obj.HttpObject;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;


@AndroidEntryPoint  // 我是被注解的 被注入的
public class MainActivity extends AppCompatActivity {
    @Inject
    HttpObject httpObject;
    @Inject
    HttpObject httpObject2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 为什么能够控制  字节码插庄
        // DaggerCOm。create.inject(this);

        Log.i("derry", httpObject.hashCode() + "");
        Log.i("derry", httpObject2.hashCode() + "");
        Toast.makeText(this, httpObject.hashCode() + "", Toast.LENGTH_SHORT).show();
        Toast.makeText(this, httpObject2.hashCode() + "", Toast.LENGTH_SHORT).show();
    }
}

File: navigation_demo/src/main/java/com/xiongtao/navigation_demo/MainActivity.kt
package com.xiongtao.navigation_demo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.NavigationUI
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {
    var bottomNavigationView: BottomNavigationView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        bottomNavigationView = findViewById(R.id.nav_view)

        val navHostFragment =
            supportFragmentManager.findFragmentById(R.id.my_nav_host_fragment) as NavHostFragment ?
        val navController = navHostFragment?.navController

        NavigationUI.setupWithNavController(bottomNavigationView!!,navController!!)

        // 为了源码分析的代码 1
        // 官网代码
        /*
        val finalHost = NavHostFragment.create(R.navigation.nav_graph_main)

        supportFragmentManager.beginTransaction()
                .replace(R.id.ll_fragment_navigation, finalHost)
                .setPrimaryNavigationFragment(finalHost)
                .commit()
         */

        // 为了源码分析的代码 2
        /*override fun onSupportNavigateUp(): Boolean {
            val fragment = supportFragmentManager.findFragmentById(R.id.my_nav_host_fragment)
            return NavHostFragment.findNavController(fragment!!).navigateUp()
        }*/

    }
}